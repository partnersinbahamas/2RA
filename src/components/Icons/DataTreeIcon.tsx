import React from 'react';
import PropTypes from 'prop-types';

import { LargeSvg, MediumSvg, SmallSvg } from './Svg';
import { TIcon } from '.';

const DataTreeIcon: React.FC<TIcon> = ({
  className,
  color,
  size,
  style,
  withBox,
}) => {
  switch (size) {
    case 'SMALL':
      return (
        <SmallSvg
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
          viewBox={withBox ? '-2 -2 24 24' : '0.8 0.8 18 18'}
          fill="transparent"
          stroke={color}
          style={style}
          className={className}
        >
          <rect x="10" y="8" width="8" height="4" rx="1" fill={color} />
          <rect x="10" y="14" width="8" height="4" rx="1" fill={color} />
          <rect
            x="2"
            y="2"
            width="8"
            height="4"
            rx="1"
            strokeWidth="2"
            strokeLinejoin="round"
          />
          <path d="M6 7V15C6 15.5523 6.44772 16 7 16H11" strokeWidth="2" />
          <path d="M6 10H11" />
        </SmallSvg>
      );
    case 'MEDIUM':
      return (
        <MediumSvg
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
          viewBox={withBox ? '-1 -1 26 26' : '0.7 0.7 22.5 22.5'}
          fill="transparent"
          stroke={color}
          strokeWidth="2"
          style={style}
          className={className}
        >
          <rect
            x="2"
            y="2"
            width="10"
            height="4"
            rx="1"
            strokeLinejoin="round"
          />
          <rect
            x="12"
            y="10"
            width="10"
            height="4"
            rx="1"
            strokeLinejoin="round"
          />
          <rect
            x="12"
            y="18"
            width="10"
            height="4"
            rx="1"
            strokeLinejoin="round"
          />
          <path d="M7 7V18C7 19.1046 7.89543 20 9 20H12" />
          <path d="M6 12H11" />
        </MediumSvg>
      );
    case 'LARGE':
    default:
      return (
        <LargeSvg
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
          viewBox={withBox ? '0 0 32 32' : '2 2 28 28'}
          fill="transparent"
          stroke={color}
          strokeWidth="2"
          style={style}
          className={className}
        >
          <rect
            x="3"
            y="3"
            width="14"
            height="6"
            rx="2"
            strokeLinejoin="round"
          />
          <rect
            x="15"
            y="13"
            width="14"
            height="6"
            rx="2"
            strokeLinejoin="round"
          />
          <rect
            x="15"
            y="23"
            width="14"
            height="6"
            rx="2"
            strokeLinejoin="round"
          />
          <path d="M10 9V24C10 25.1046 10.8954 26 12 26H15" />
          <path d="M10 16L15 16" />
        </LargeSvg>
      );
  }
};

DataTreeIcon.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  style: PropTypes.any,
  withBox: PropTypes.bool,
};

DataTreeIcon.defaultProps = {
  className: '',
  color: 'currentcolor',
  size: 'LARGE',
  style: {},
  withBox: false,
};

export default DataTreeIcon;
