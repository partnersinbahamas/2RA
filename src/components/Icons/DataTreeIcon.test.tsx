import React from 'react';
import { render } from '@testing-library/react';
import DataTreeIcon from './DataTreeIcon';

describe('DataTreeIcon', () => {
  it('renders small', () => {
    const { container } = render(<DataTreeIcon size="SMALL" />);
    expect(container).toMatchSnapshot();
  });

  it('renders medium', () => {
    const { container } = render(<DataTreeIcon size="MEDIUM" />);
    expect(container).toMatchSnapshot();
  });

  it('renders large', () => {
    const { container } = render(<DataTreeIcon size="LARGE" />);
    expect(container).toMatchSnapshot();
  });

  it('renders default', () => {
    const { container } = render(<DataTreeIcon />);
    expect(container).toMatchSnapshot();
  });
});
