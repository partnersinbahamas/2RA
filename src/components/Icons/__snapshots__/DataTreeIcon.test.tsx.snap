// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTreeIcon renders default 1`] = `
<div>
  <svg
    class="sc-hLQSwg hewGWh"
    fill="transparent"
    stroke="currentcolor"
    stroke-width="2"
    viewBox="2 2 28 28"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink"
  >
    <rect
      height="6"
      rx="2"
      stroke-linejoin="round"
      width="14"
      x="3"
      y="3"
    />
    <rect
      height="6"
      rx="2"
      stroke-linejoin="round"
      width="14"
      x="15"
      y="13"
    />
    <rect
      height="6"
      rx="2"
      stroke-linejoin="round"
      width="14"
      x="15"
      y="23"
    />
    <path
      d="M10 9V24C10 25.1046 10.8954 26 12 26H15"
    />
    <path
      d="M10 16L15 16"
    />
  </svg>
</div>
`;

exports[`DataTreeIcon renders large 1`] = `
<div>
  <svg
    class="sc-hLQSwg hewGWh"
    fill="transparent"
    stroke="currentcolor"
    stroke-width="2"
    viewBox="2 2 28 28"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink"
  >
    <rect
      height="6"
      rx="2"
      stroke-linejoin="round"
      width="14"
      x="3"
      y="3"
    />
    <rect
      height="6"
      rx="2"
      stroke-linejoin="round"
      width="14"
      x="15"
      y="13"
    />
    <rect
      height="6"
      rx="2"
      stroke-linejoin="round"
      width="14"
      x="15"
      y="23"
    />
    <path
      d="M10 9V24C10 25.1046 10.8954 26 12 26H15"
    />
    <path
      d="M10 16L15 16"
    />
  </svg>
</div>
`;

exports[`DataTreeIcon renders medium 1`] = `
<div>
  <svg
    class="sc-dmyCSP dRcSrL"
    fill="transparent"
    stroke="currentcolor"
    stroke-width="2"
    viewBox="0.7 0.7 22.5 22.5"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink"
  >
    <rect
      height="4"
      rx="1"
      stroke-linejoin="round"
      width="10"
      x="2"
      y="2"
    />
    <rect
      height="4"
      rx="1"
      stroke-linejoin="round"
      width="10"
      x="12"
      y="10"
    />
    <rect
      height="4"
      rx="1"
      stroke-linejoin="round"
      width="10"
      x="12"
      y="18"
    />
    <path
      d="M7 7V18C7 19.1046 7.89543 20 9 20H12"
    />
    <path
      d="M6 12H11"
    />
  </svg>
</div>
`;

exports[`DataTreeIcon renders small 1`] = `
<div>
  <svg
    class="sc-guDLey kBBjNb"
    fill="transparent"
    stroke="currentcolor"
    viewBox="0.8 0.8 18 18"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink"
  >
    <rect
      fill="currentcolor"
      height="4"
      rx="1"
      width="8"
      x="10"
      y="8"
    />
    <rect
      fill="currentcolor"
      height="4"
      rx="1"
      width="8"
      x="10"
      y="14"
    />
    <rect
      height="4"
      rx="1"
      stroke-linejoin="round"
      stroke-width="2"
      width="8"
      x="2"
      y="2"
    />
    <path
      d="M6 7V15C6 15.5523 6.44772 16 7 16H11"
      stroke-width="2"
    />
    <path
      d="M6 10H11"
    />
  </svg>
</div>
`;
